src/lib.rs

```diff
@@ -460,29 +463,39 @@ fn expand_template(
     name: &ProjectName,
     dir: &Path,
     template_values: &HashMap<String, toml::Value>,
-    template_config: Config,
+    mut template_config: Config,
     args: &Args,
 ) -> Result<()> {
     let crate_type: CrateType = args.into();
-    let template = template::create_liquid_object(name, &crate_type, args.force)?;
-    let template = project_variables::fill_project_variables(template, &template_config, |slot| {
-        let provided_value = template_values.get(&slot.var_name).and_then(|v| v.as_str());
-        if provided_value.is_none() && args.silent {
-            anyhow::bail!(ConversionError::MissingPlaceholderVariable {
-                var_name: slot.var_name.clone()
-            })
-        }
-        interactive::variable(slot, provided_value)
-    })?;
-    let template = add_missing_provided_values(template, template_values)?;
-
-    let (template_cfg, template) = merge_conditionals(template_config, template, args)?;
+    let liquid_object = template::create_liquid_object(name, &crate_type, args.force)?;
```

Comment:
question: I would prefer to not have such random renaming of variables when there is no need for chnage, the names have all been choosen on purpose and I'd kindly ask you to respect the authors naming decisions and rather adapt your mental model. 

Ofc if you think that something is just pure wrong or nonsense then I'd suggest to keep those discussions into separated PRs, that would simplify the review and keep things pretty focused on the feature.

Reply:
If you refer to the `liquid_object` variable - I confess to having renamed it from the original name `template`.

Having read, and re-read, your comment a few times (because I am unsure what to make of it) - I think have a fair grasp of the issue at hand; it's about the fact I did a rename as part of this commit/PR, and not the actual rename - right?
**For that I am sorry**; I did not realize I should restrict each commit/PR that strictly - not used to that.

If on the other hand it's about the actual rename, here's my reasoning (and original reply prior to catching the alternate meaning/reason for the comment)...

**Reason for rename:** back in August, the `template::substiture(...)` function was renamed (by me, again!) to `create_liquid_object(...)` - at that time because *the function didn't do any substitution at all*, it just created and prepared a `Object` from the `liquid` crate - litterally.

What I did not notice at the time was the variable in question. It is not a template, nor does it represent one. It is a object for holding template values when `liquid` eventually does substitution in the actual template files - in the `walk_dir(...)` function.

Hence I renamed it to reflect this discovery.

**If I was wrong, or anyone fells hurt, I appologise here and now** - but I do feel the original name didn't fit the actual use anymore.

If you want me to rename it back, please just say so.


For the record: I myself *do not care if anyone renames or otherwise changes any part of any code I have ever written*. I actually expect the other contributors to do so, in order for them to better understand the code and/or for the code to be in the best shape it can be at any point in time.

Reply:
> For that I am sorry; I did not realize I should restrict each commit/PR that strictly - not used to that.

no worries, no feelings hurt. All good. If renaming make sense, and your provided reasoning does make sense, then go for it. I'm not tied to names (only if I spend a fair amount of time to express a concept and this expression level would decrease because of the rename).

So sub notion was about focus on PRs, so this PR is already quite a commit 💣  which is great because of the nice feature, but the level of noise (coming from random function / variable renaming) might be (if possible) kept low.

So having that said, I totally can support this particular rename.